### PR TITLE
Add more information to user enforced consistency

### PR DESCRIPTION
**Pull request by premkumr**
- Added a Caution for the user to be wary of issues.
- Made a note on backfilling

### CODE REVIEW

**Review by KavyaShivashankar — CHANGES_REQUESTED** [review]
Index backfill does not automatically occur due the value set on this flag ```disable_index_backfill_for_non_txn_tables=true```.   Once the value of this flag is set to ```false```,  automatic backfill happens at the time of Index creation.  

 Backfill operation has to be manually initiated for index created during the time this flag had a value ```true```

**Comment by premkumr** [comment]
@KavyaShivashankar  - please verify!

File: docs/content/stable/api/ycql/ddl_create_index.md [hunk 1]
```diff
@@ -113,6 +117,10 @@ CREATE INDEX ON orders (warehouse)
 When using an index without transactions enabled, it is the responsibility of the application to retry any insert/update/delete failures to make sure that the table and index are in sync.
 {{< /warning >}}
 
+{{< warning title="Syncing table and index">}}
+When using an index without transactions enabled, it is the responsibility of the application to retry any insert/update/delete failures to make sure that the table and index are in sync. <br><br>Also, if the index is created after data has been added to the table, the index may **not** be backfilled automatically depending on the setting of the gflag `disable_index_backfill_for_non_txn_tables`. If it is set to `true`, then it will be the responsibility of the user to trigger a backfill using the [yb-admin backfill_indexes_for_table](../../../admin/yb-admin/#backfill-indexes-for-table) which will trigger the backfill after a small delay of about a minute. This delay is controlled by the gflag: `index_backfill_upperbound_for_user_enforced_txn_duration_ms`.
```
Comment (aishwarya24) [inline docs/content/stable/api/ycql/ddl_create_index.md:121]:
```suggestion
When using an index without transactions enabled, it is the responsibility of the application to retry any insert/update/delete failures to make sure that the table and index are in sync. 

Also, if the index is created after data has been added to the table, the index may **not** be backfilled automatically depending on the setting of the `disable_index_backfill_for_non_txn_tables` flag. If set to `true`, then it is the responsibility of the user to trigger a backfill using the [yb-admin backfill_indexes_for_table](../../../admin/yb-admin/#backfill-indexes-for-table) command, which will trigger the backfill after a small delay of about a minute. This delay is controlled by the `index_backfill_upperbound_for_user_enforced_txn_duration_ms` flag.
```

File: docs/content/preview/api/ycql/ddl_create_index.md [hunk 2]
```diff
@@ -103,6 +103,10 @@ When an index is created on an existing table, YugabyteDB will automatically bac
 
 ### User enforced consistency
```
Comment (ddhodge) [inline docs/content/preview/api/ycql/ddl_create_index.md:104]:
```suggestion
### User-enforced consistency
```

File: docs/content/stable/api/ycql/ddl_create_index.md [hunk 3]
```diff
@@ -101,6 +101,10 @@ When an index is created on an existing table, YugabyteDB will automatically bac
 
 ### User enforced consistency
```
Comment (ddhodge) [inline docs/content/stable/api/ycql/ddl_create_index.md:102]:
```suggestion
### User-enforced consistency
```

**Review by ddhodge — APPROVED** [review]

**Pull request merged**